Data/task.py

```diff
@@ -1,9 +1,30 @@
+from Data.future import Future
+import api_pb2
+
 class Task:
-    
+    # self.taskData is a list of future or/and bytes
     def __init__(self, taskId, taskDefintion, taskData):
         self.taskId = taskId
         self.taskDefintion = taskDefintion
-        self.taskData = taskData
+        if len(taskData)!=0 and not isinstance(taskData[0], api_pb2.TaskDataParam):
+            self.taskData = taskData
```

Comment:
in which case, will this be called

Reply:
When you pass taskData which is a list of Data.Future and/or bytes.
The constructor handles both proto object (List of TaskDataParam) / list of future and/or bytes

Reply:
> When you pass taskData which is a list of Data.Future and/or bytes.
I thought else part is handling this.

What is the purpose of if true?

Reply:
else part handles the conversion of proto taskData (list of TaskDataParam objects) to self.taskData (list of Data.Future and/or bytes). 

If part just saves the incoming taskData param in the constructor to self.taskData, since it's already a list of future and bytes (needs no conversion)